tests/e2e/streams/common.py

```diff
@@ -255,31 +245,56 @@ def call_operation(counter):
         if operation_proc.is_alive():
             operation_proc.terminate()
 
+
 def test_start_checked_stream_after_timeout(connection, stream_creator):
     cursor = connection.cursor()
-    execute_and_fetch_all(
-        cursor,
-        stream_creator('test_stream')
-    )
+    execute_and_fetch_all(cursor, stream_creator("test_stream"))
 
     timeout_ms = 2000
 
     def call_check():
-        execute_and_fetch_all(
-            connect().cursor(),
-            f"CHECK STREAM test_stream TIMEOUT {timeout_ms}")
+        execute_and_fetch_all(connect().cursor(), f"CHECK STREAM test_stream TIMEOUT {timeout_ms}")
 
     check_stream_proc = Process(target=call_check, daemon=True)
 
     start = time.time()
     check_stream_proc.start()
-    assert timed_wait(
-        lambda: get_is_running(
-            cursor, "test_stream"))
+    assert timed_wait(lambda: get_is_running(cursor, "test_stream"))
     start_stream(cursor, "test_stream")
     end = time.time()
 
-    assert (end - start) < 1.3 * \
-        timeout_ms, "The START STREAM was blocked too long"
+    assert (end - start) < 1.3 * timeout_ms, "The START STREAM was blocked too long"
     assert get_is_running(cursor, "test_stream")
     stop_stream(cursor, "test_stream")
+
+
+def test_start_stream_with_batch_limit(connection, stream_creator, messages_sender):
```

Comment:
I would add one or two more tests to make sure calling `CHECK` while `START STREAM ... LIMIT_BATCHES` is resulting in the correct exception and not a crash. The second test would be the other way around (calling `START STREAM ... LIMIT_BATCHES` while `CHECK` is running), but it might be done in one tests with parameters.